scalafix-core/src/main/scala/scalafix/rewrite/NoUnusedImports.scala

```diff
@@ -0,0 +1,54 @@
+package scalafix
+package rewrite
+
+import scala.meta._
+import scala.annotation.tailrec
+
+case class NoUnusedImports(mirror: Mirror) extends SemanticRewrite(mirror) {
```

Comment:
One challenge with imports is that I suspect many rewrites will want to modify them so there's a risk of patch conflicts. My eventual plan is to take a holistic approach towards imports management and bake it into the Patch system. Instead of eagerly removing/adding import tokens inside rewrites, we can collect high-level patches (e.g., add/remove import, rename) inside rewrites and evaluate them to token patches as late as possible. This might make it easier to compose different rewrites.

My dream is that rewrite authors won't have to worry too much about low-level details of imports (because they're kinda complex in Scala) and scalafix users can configure how they like to organize their imports https://github.com/scalacenter/scalafix/blob/996f30ff87ad9442c53a06ccf505ac6e845b9672/scalafix-core/src/main/scala/scalafix/config/ImportsConfig.scala

However, to begin with, I think there is value in having a custom rewrite only to remove unused imports. My experience writing an OrganizeImports step was that it's quite tricky to get right, so it may take a few iterations until we get there and in the meantime a RemoveUnusedImports rewrite might turn out to be useful enough.